sevenzipjbindings/src/main/java/com/mucommander/sevenzipjbindings/multivolume/SevenZipMultiVolumeCallbackHandler.java

```diff
@@ -0,0 +1,92 @@
+package com.mucommander.sevenzipjbindings.multivolume;
+
+import com.mucommander.commons.file.AbstractFile;
+import com.mucommander.commons.file.FileFactory;
+import com.mucommander.sevenzipjbindings.SignatureCheckedRandomAccessFile;
+import net.sf.sevenzipjbinding.*;
+import net.sf.sevenzipjbinding.impl.RandomAccessFileInStream;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.Closeable;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.RandomAccessFile;
+import java.util.HashMap;
+import java.util.Map;
+
+public class SevenZipMultiVolumeCallbackHandler implements IArchiveOpenVolumeCallback, ICryptoGetTextPassword, Closeable {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(SevenZipMultiVolumeCallbackHandler.class);
+
+    private Map<String, IInStream> fileCache = new HashMap<>();
+
+    private final AbstractFile firstFile;
+
+    private final byte[] signature;
+
+    private final String password;
+
+    private boolean isFirstFile = true;
+
+    public SevenZipMultiVolumeCallbackHandler(byte[] signature, AbstractFile firstFile, String password) {
+        this.signature = signature;
+        this.firstFile = firstFile;
+        this.password = password;
+    }
+
+    @Override
+    public Object getProperty(PropID propID) throws SevenZipException {
+        switch (propID) {
+            case NAME:
+                return firstFile.getAbsolutePath();
+        }
+        return null;
+    }
+
+    @Override
+    public IInStream getStream(String filename) throws SevenZipException {
+        try {
+            IInStream stream = fileCache.get(filename);
+            if (stream != null) {
+                stream.seek(0, ISeekableStream.SEEK_SET);
+            } else {
+                AbstractFile abstractFile = FileFactory.getFile(filename);
```

Comment:
it's not used when we're not working on the first volume so please inline in the block it's used in

Reply:
nice catch - fixed